templates/server/src/main/resources/archetype-resources/core/pom.xml

```diff
@@ -122,50 +122,50 @@
 
     <!-- Database -->
     <dependency>
-#if ($dbType == 'h2')
+      #if ($dbType == 'h2')
```

Comment:
it is great that you do auto-formatting now.
Unfortunately in the archetype with velocity macros this is not so great as the extra whitespaces in front of macros will be written to the output. Therefore the generated POM will be messed with whitespaces. Can you have a look at the resulting generated POM?